tasks/Functions-client/buildOffchainSecrets.js

```diff
@@ -0,0 +1,115 @@
+  .setAction(async (taskArgs) => {
+    if (network.name === "hardhat") {
+      throw Error(
+        'This command cannot be used on a local hardhat chain.'
+      )
+    }
+
+    const requestConfig = require("../../Functions-request-config")
+
+    // Verify that perNodeSecrets and/or secrets is correctly specified in the config
+    if (requestConfig.perNodeSecrets && !Array.isArray(requestConfig.perNodeSecrets)) {
+      throw Error('perNodeSecrets is not correctly specified in config file.  It must be an array of objects.')
+    }
+
+    if (requestConfig.secrets && typeof requestConfig.secrets !== 'object') {
+      throw Error('secrets object is not correctly specified in config file.  It must be an object.')
+    }
+
+    if (
+      (!requestConfig.perNodeSecrets || requestConfig.perNodeSecrets.length === 0)
+      && (!requestConfig.secrets || Object.keys(requestConfig.secrets).length === 0)
+    ) {
+      throw Error('Neither perNodeSecrets nor secrets is specified')
+    }
+
+    const defaultSecretsObjectKeys = requestConfig.secrets
+      ? JSON.stringify(Object.keys(requestConfig.secrets).sort())
+      : undefined
+
+    if (!requestConfig.secrets || defaultSecretsObjectKeys.length === 0) {
+      console.log("\nWARNING: No default `secrets` provided.  If DON membership changes, the new node will not be able to process requests.\n")
+    }
+
+    if (requestConfig.perNodeSecrets && requestConfig.perNodeSecrets.length > 0) {
+      const firstPerNodeSecretsKeys = JSON.stringify(Object.keys(requestConfig.perNodeSecrets[0]).sort())
+
+      for (const assignedSecrets of requestConfig.perNodeSecrets) {
+        if (typeof assignedSecrets !== 'object') {
+          throw Error('perNodeSecrets is not correctly specified in config file.  It must be an array of objects.')
+        }
+  
+        if (Object.keys(assignedSecrets).length === 0) {
+          throw Error('In the config file, perNodeSecrets contains an empty object.')
+        }
+  
+        const secretsObjectKeys = JSON.stringify(Object.keys(assignedSecrets).sort())
+  
+        if (
+          requestConfig.secrets
+          && defaultSecretsObjectKeys.length > 0
+          && defaultSecretsObjectKeys !== secretsObjectKeys
+        ) {
+          throw Error('In the config file, not all objects in `perNodeSecrets` have the same object keys as default `secrets`. (The values can be different, but the keys should be the same between all objects.)')
+        }
+
+        if (firstPerNodeSecretsKeys !== secretsObjectKeys) {
+          throw Error('In the config file, not all objects in `perNodeSecrets` have the same object keys. (The values can be different, but the keys should be the same between all objects.)')
+        }
+      }
+    }
+
+    console.log(`Using public keys from FunctionsOracle contract ${networkConfig[network.name]['functionsOracle']} on network ${network.name}`)
+    const OracleContract = await ethers.getContractFactory("FunctionsOracle")
```

Comment:
This should be called *Factory. It's too confusing to have two vars that differ only by casing. It's JS after all, not Go ;)

Reply:
lol its also confusing that one of our contracts is called OracleFactory too.  However I will make the change here.